pkg/collector/deployment.go

```diff
@@ -48,7 +47,7 @@ func Deployment(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTele
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
 					Labels:      labels,
-					Annotations: podAnnotations,
+					Annotations: annotations,
```

Comment:
Won't this introduce a regression where you can no longer set specific pod annotations? I rely on exactly this currently in a few places.

Reply:
correct , I missed this 👍🏼

Reply:
Okay, so, to avoid this kind of regression, I would propagate the annotations into podAnnotations. WDYT?

Reply:
I think it would be fine to append the annotations in to the pod annotations. How would you handle conflicts?

Reply:
WDYT to take the Deployment's annotations as precedence?

Reply:
@pavolloffay , @jaronoff97 ?

Reply:
@yuriolisa i think the pod annotations should win, otherwise there would be no way to override something on the deployment.